SchoolManagerSystem.Service/Authentications/Implementations/AuthServices.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.AspNetCore.Identity;
+using SchoolManagerSystem.Common.DTOs;
+using SchoolManagerSystem.Model.Entities;
+using SchoolManagerSystem.Service.Authentications.Interfaces;
+using SchoolManagerSystem.Service.Token.Interfaces;
+using System;
+using System.Threading.Tasks;
+
+namespace SchoolManagerSystem.Service.Authentications.Implementations
+{
+	public class AuthServices : IAuthServices
+	{
+		private readonly UserManager<ApplicationUser> _userManager;
+		private readonly RoleManager<IdentityRole> _roleManager;
+		private readonly IToken _token;
+
+		public AuthServices(UserManager<ApplicationUser> userManager, RoleManager<IdentityRole> roleManager, IToken token)
+		{
+			_userManager = userManager;
+			_roleManager = roleManager;
+			_token = token;
+		}
+
+
+		public async Task<UserRegistrationResponse> Register(UserRegistrationRequest registerRequest, string isRole)
+		{
+			var user = new ApplicationUser
+			{
+				FirstName = registerRequest.FirstName,
+				LastName = registerRequest.LastName,
+				Email = registerRequest.Email,
+				UserName = registerRequest.UserName,
+				PasswordHash = registerRequest.Password,
+				EmailConfirmed = true
+			};
+			var results = await _userManager.CreateAsync(user, registerRequest.Password);
+			if (!results.Succeeded)
+			{
+				var errors = string.Empty;
+				foreach (var error in results.Errors)
+				{
+					errors = error.Description + Environment.NewLine;
+				}
+				throw new MissingFieldException(errors);
+			}
+
+			var isRoleTheSame = isRole.Equals("Principal");
```

Comment:
Why are you adding a role when registering a user.. 
all your implementation regarding your role in this class violates the solid principle.
role in the system should be assigned to a user base on criteria.

Did you test this implementation and it works for you.. so at the point of registration you want a user to chose their role or add a new role.